v2/pkg/scan/arp.go

```diff
@@ -16,9 +16,14 @@ func init() {
 
 // ArpRequestAsync asynchronous to the target ip address
 func ArpRequestAsync(s *Scanner, ip string) {
+	srcAddr := net.HardwareAddr{}
```

Comment:
The network interface should be calculated with `s.Router.Route(ip)`, see for example how it's handled for ICMP https://github.com/projectdiscovery/naabu/blob/60f4b5dbd58a39159937d3f40503195663c69b49/v2/pkg/scan/icmp.go#L102

ARP packets would be rejected by the network interface without a valid source MAC